src/services/Odin.Services/LinkMetaExtractor/LinkMeta.cs

```diff
@@ -31,58 +33,113 @@ public static LinkMeta FromMetaData(Dictionary<string, object> meta, string url)
-        {
-            return value1.ToString() ?? throw exception;
-        }
-        else if (meta.TryGetValue("twitter:title", out var value2))
+        if (s != null && s.Length > maxLength)
+            s = s.Substring(0, maxLength - 3) + "...";
+
+        return s;
+    }
+
+    private static string? GetTitle(Dictionary<string, object> meta)
+    {
+        foreach (var key in new[] { "title", "og:title", "twitter:title" })
         {
-            return value2.ToString() ?? throw exception;
+            if (meta.TryGetValue(key, out var value) && !string.IsNullOrWhiteSpace(value?.ToString()))
+            {
+                return value.ToString();
+            }
         }
-        throw exception;
+
+        return null;
     }
-    
 
-    private static string? GetDescription(Dictionary<string, object> meta)
+
+    public static string? GetDescription(Dictionary<string, object> meta)
     {
+        var keys = new[] { "description", "og:description", "twitter:description" };
 
-        if (meta.TryGetValue("description", out var value))
+        foreach (var key in keys)
         {
-            return value.ToString();
+            if (meta.TryGetValue(key, out var value) && !string.IsNullOrWhiteSpace(value?.ToString()))
+            {
+                return value.ToString();
+            }
         }
-        else if (meta.TryGetValue("og:description", out var value1))
-        {
-            return value1.ToString();
-        }
-        else if (meta.TryGetValue("twitter:description", out var value2))
+
+        return null;
+    }
+
+    /// <summary>
+    /// Extracts the image URL or embedded image data from metadata.
+    /// The function prioritizes the keys "og:image" and "twitter:image" to locate the image.
+    /// If a valid URL is found, it is returned. If the value contains embedded image data
+    /// (e.g., a "data:image" URI), it validates the MIME type and limits the size.
+    /// Returns null if no valid image URL or embedded data is found.
+    /// </summary>
+    /// <param name="meta">A dictionary of metadata keys and values.</param>
+    /// <returns>A string containing the image URL or sanitized embedded image data, or null if none is found.</returns>
+    public static string? GetImageUrl(Dictionary<string, object> meta)
+    {
+        var candidates = new[] { "og:image", "twitter:image" };
+
+        foreach (var key in candidates)
         {
-            return value2.ToString();
+            if (meta.TryGetValue(key, out var value) && !string.IsNullOrWhiteSpace(value?.ToString()))
+            {
+                var imageData = value.ToString();
+
+                if (imageData == null)
+                    continue;
+
+                // Find an embedded image
+                if (IsValidEmbeddedImage(imageData))
+                    return imageData;
+
+                // Otherwise, validate it as a safe URL
+                if (LinkMetaExtractor.IsUrlSafe(imageData))
```

Comment:
Do we still want to return the imageUrl? wouldn't that link may contain trackers?